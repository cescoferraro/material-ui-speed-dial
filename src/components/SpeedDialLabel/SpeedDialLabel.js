// @flow
import React from 'react';
import { withStyles } from 'material-ui/styles';

import styles from './SpeedDialLabel.styles';

// Types
import type { SpeedDialLabelPropsType } from '../../../material-ui-speed-dial.js.flow';
import type { ClassesType } from '../../types/styles';

export type SpeedDialLabelWithStylePropsType = {
	classes: ClassesType,
};

export const SpeedDialLabel = ({
	className,
	classes,
	text,
}: SpeedDialLabelWithStylePropsType & SpeedDialLabelPropsType): React$Element<*> => {
	return <span className={`${className} ${classes.root}`}>{text}</span>;
};

// $FlowFixMe
export default (withStyles(styles)(SpeedDialLabel): React$Element<*>);
