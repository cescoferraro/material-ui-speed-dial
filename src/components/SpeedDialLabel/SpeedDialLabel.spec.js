import React from 'react';
import { shallow } from 'enzyme';
import { SpeedDialLabel } from './SpeedDialLabel';
import styles from './SpeedDialLabel.styles';

const props = {
	classes: styles,
	className: 'className',
	text: 'text',
};

describe('SpeedDialLabel', () => {
	it('snapshot: default', () => {
		const _props = { ...props };
		const tree = shallow(<SpeedDialLabel {..._props} />);
		expect(tree).toMatchSnapshot();
	});
	it('styles', () => {
		expect(
			typeof styles({
				typography: { body1: {} },
				palette: { background: { paper: '' } },
				shadows: [0, 1, 2, 3, 4, 5],
			})
		).toBe('object');
	});
});
