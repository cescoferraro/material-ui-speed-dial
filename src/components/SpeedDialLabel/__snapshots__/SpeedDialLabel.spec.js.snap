// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedDialLabel snapshot: default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    className="className"
    classes={[Function]}
    text="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "text",
      "className": "className undefined",
    },
    "ref": null,
    "rendered": "text",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "text",
        "className": "className undefined",
      },
      "ref": null,
      "rendered": "text",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
